system/lib/wasmfs/wasmfs.cpp

```diff
@@ -101,6 +102,46 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+long __syscall_fstat64(long fd, long buf) {
+  // Release FileTable lock after accessing desired open file.
+  std::shared_ptr<File> file = [&] {
+    auto fileTable = FileTable::get();
+    return fileTable[fd]->get().getFile();
+  }();
+
+  struct stat* buffer = (struct stat*)buf;
+
+  auto fileInfo = file->get();
+
+  if (file->is<Directory>()) {
+    buffer->st_size = 4096;
```

Comment:
Is this constant copied from somewhere?

Reply:
Yeah they are copied from the JS code.

Reply:
I see, thanks. I don't see a comment explaining why, there. How about at least adding one here, that this matches the previous JS behavior, but is not accurate.

Reply:
Or we can expand that existing comment to say that the specific values are chosen to match the existing library_fs.js values.

Reply:
Sounds good to me